src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -306,6 +308,24 @@ protected async Task<IActionResult> ExecuteQuery<TPostBodyResource>(TResource re
                     }
                     catch (Exception ex)
                     {
+                        if (ex.InnerException is KustoTenantListEmptyException)
+                        {
+                            if (
+                                   (
+                                   // Resource is ASE
+                                    resource as IResource is HostingEnvironment env && (env.TenantIdList != null && !env.TenantIdList.Any())
```

Comment:
No a big deal but it looks like your using this logic in two locations of this file, I recommend just creating a method that does this logic for you to avoid any risk of mismatched logic and etc.